### PR TITLE
toggle-button

### LINKED ISSUE #349: [COMPONENT] Toggle
**Describe the solution you'd like**
Toggles change the state of a single option. Toggles can be switched on or off by pressing or swiping them. They can also be checked programmatically by setting the checked property.

**Additional context**
https://ionicframework.com/docs/api/toggle
https://docs.nativescript.org/angular/ui/ng-ui-widgets/switch

ios and web looks like ios 
android looks like android

Colors are coming from kirby

Se også her for at se hvordan vi kommer til at bruge det:
https://zpl.io/beKAX3B

### Kick Off:
- [x] Accept Feature incl. name of component (DSL)
- [x] Update Kirby Component Status and publish to [Kirby Cookbook](https://cookbook.kirby.design)
### Code:
- [x] Create Feature Branch
- [x] Implement Skeleton Component:
    - [x] Web implementation
    - [x] {N} implementation
- [x] [Implement Examples and Showcase](https://cookbook.kirby.design/home/showcase/button)
- [x] Add link to showcase on Kirby Component Status (`component-status-items.ts => code.cookbookUrl`)
- [x] Also remove `githubIssueNo` and set `status` to `ItemCodeStatus.ready`
- [x] Push commits to GitHub
### Review:
- [x] Open a pull request (remember to add `closes #issueno` to the description) and ask @kirbydesign/kirby-guild for review
- Merge to master and delete branch

### PR DESCRIPTION
**Pull request by lydemann**
This PR closes #349  (reference issue number here)

Toggle button
image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Added toggle button

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: apps/cookbook/src/app/examples/examples.module.ts [hunk 1]
```diff
@@ -17,34 +17,25 @@ import { ChartExampleModule } from './chart-example/chart-example.module';
 import { ProgressCircleExampleModule } from './progress-circle-example/progress-circle-example.module';
 import { AvatarExampleModule } from './avatar-example/avatar-example.module';
 import { FlagExampleModule } from './flag-example/flag-example.module';
+import { ToggleButtonExampleModule } from './toggle-button-example/toggle-button-example.module';
+
+const IMPORTS = [
```
Comment (lydemann) [inline apps/cookbook/src/app/examples/examples.module.ts:22]: Let's get serious and clean some imports

File: apps/cookbook/src/app/showcase/toggle-button-showcase/toggle-button-showcase.component.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Component } from '@angular/core';
+
+import { ShowcaseProperty } from '../../shared/showcase-properties/showcase-property';
+declare var require: any;
+
+@Component({
+  templateUrl: './toggle-button-showcase.component.html',
+})
+export class ToggleButtonShowcaseComponent {
+  exampleHtml: string = require('!raw-loader!../../examples/toggle-button-example/toggle-button-example.component.html')
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/toggle-button-showcase/toggle-button-showcase.component.ts:10]: We expose the template through a config object now - e.g. see `apps\cookbook\src\app\examples\segmented-control-example\default\default.ts`

File: apps/cookbook/src/app/examples/toggle-button-example/toggle-button-example.component.html [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<kirby-toggle-button [checked]="true" (checkedChange)="onCheckedChange($event)">
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/toggle-button-example/toggle-button-example.component.html:1]: We expose the template through a config object now - e.g. see apps\cookbook\src\app\examples\segmented-control-example\default\default.ts

File: apps/cookbook/src/app/examples/toggle-button-example/toggle-button-example.component.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'cookbook-toggle-button-example',
+  templateUrl: './toggle-button-example.component.html',
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/toggle-button-example/toggle-button-example.component.ts:5]: We expose the template through a config object now - e.g. see apps\cookbook\src\app\examples\segmented-control-example\default\default.ts

File: apps/cookbook/src/app/showcase/toggle-button-showcase/toggle-button-showcase.component.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<div class="example">
+  <cookbook-toggle-button-example></cookbook-toggle-button-example>
+  <cookbook-code-viewer [html]="exampleHtml"></cookbook-code-viewer>
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/toggle-button-showcase/toggle-button-showcase.component.html:3]: We now embed the example inside `cookbook-example-viewer` , e.g. see `apps\cookbook\src\app\showcase\segmented-control-showcase\segmented-control-showcase.component.html`

File: libs/designsystem/src/lib/components/toggle-button/toggle-button.component.scss [hunk 6]
```diff
@@ -0,0 +1,10 @@
+@import '../../scss/utils';
+
+:host {
+  @each $color-name, $color-value in $notification-colors {
+    :host(.#{$color-name}) {
+      --kirby-button-background-color: #{get-color($color-name)};
+      --kirby-button-color: #{get-color($color-name + '-contrast')};
+    }
+  }
+}
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toggle-button/toggle-button.component.scss:10]: Won't work - move to `button.component.scss`

File: libs/designsystem/src/lib/helpers/color-helper.ts [hunk 7]
```diff
@@ -143,3 +143,9 @@ export interface KirbyColor extends Color {
   shade: Color;
   contrast: Color;
 }
+
+export enum NotificationColor {
+  success = 'success',
+  warning = 'warning',
+  danger = 'danger',
+}
```
Comment (jakobe) [inline libs/designsystem/src/lib/helpers/color-helper.ts:151]: :fire: Don't repeat yourself - this already exists in `ColorHelper` via `color-helper.styles` auto-generated from scss
```suggestion
```

File: libs/designsystem/src/lib/components/toggle-button/toggle-button.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+import { ToggleButtonComponent } from './toggle-button.component';
+
+describe('ToggleButtonComponent', () => {
+  let spectator: Spectator<ToggleButtonComponent>;
+  const createComponent = createComponentFactory(ToggleButtonComponent);
+
+  beforeEach(() => (spectator = createComponent()));
+
+  it('should should toggle checked state on click', (done) => {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toggle-button/toggle-button.component.spec.ts:12]: Consider testing for both the event emitter (checkChanged) and the state (checked)

**Review by jakobe — CHANGES_REQUESTED** [review]
See comments

File: libs/designsystem/src/lib/helpers/color-helper.ts (hunk 7 above)
Comment (lydemann) [reply libs/designsystem/src/lib/helpers/color-helper.ts:151]: No an enum like this doesn't exist to use for type safety doesn't exist.

File: libs/designsystem/src/lib/components/toggle-button/toggle-button.component.spec.ts (hunk 8 above)
Comment (lydemann) [reply libs/designsystem/src/lib/components/toggle-button/toggle-button.component.spec.ts:12]: Both get covered here?

Comment (jakobe) [reply libs/designsystem/src/lib/components/toggle-button/toggle-button.component.spec.ts:12]: Nope, you only expect the event being emitted, you don't actually test the state (`checked === false`)

File: libs/designsystem/src/lib/components/button/button.component.scss [hunk 9]
```diff
@@ -237,3 +237,17 @@
 :host-context(kirby-dropdown) {
   justify-content: space-between;
 }
+
+.danger {
+  --kirby-button-background-color: yellow;
+  --kirby-button-color: white;
+}
+
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/button/button.component.scss:245]: 💩 :)
```suggestion
```

File: libs/designsystem/src/lib/components/button/button.component.spec.ts [hunk 10]
```diff
@@ -4,6 +4,7 @@ import { createHostFactory, SpectatorHost } from '@ngneat/spectator';
 import { DesignTokenHelper } from '../../helpers/design-token-helper';
 import { IconComponent } from '../icon/icon.component';
 import { ButtonComponent } from './button.component';
+import { ColorHelper, KirbyColor, NotificationColor } from '../../helpers';
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/button/button.component.spec.ts:7]: Is this used anywhere??

File: libs/designsystem/src/lib/components/toggle-button/toggle-button.module.ts [hunk 11]
```diff
@@ -2,9 +2,8 @@ import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 
 import { ToggleButtonComponent } from './toggle-button.component';
-import { ToggleButtonThemeColorDirective } from './directives/toggle-button-color.directive';
 
-const DECLARATIONS = [ToggleButtonComponent, ToggleButtonThemeColorDirective];
+const DECLARATIONS = [ToggleButtonComponent];
 
 @NgModule({
   declarations: [...DECLARATIONS],
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/toggle-button/toggle-button.module.ts]: Does it make sense to have a module for 1 component? We don't do that elsewhere...

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/helpers/color-helper.ts (hunk 7 above)
Comment (jakobe) [reply libs/designsystem/src/lib/helpers/color-helper.ts:151]: Then add it to the autogeneration of  `color-helper.styles`

File: libs/designsystem/src/lib/components/toggle-button/toggle-button.module.ts (hunk 11 above)
Comment (lydemann) [reply libs/designsystem/src/lib/components/toggle-button/toggle-button.module.ts]: Does it make sense to create a class that only contains one public method, same thing...

I believe every ui component should have it's own module.
Check Angular Material:
https://github.com/angular/components/blob/master/src/material/button/button-module.ts

They create a module for each component, so everything necessary for that UI component is encapsulated in the module.

File: libs/designsystem/src/lib/helpers/color-helper.ts (hunk 7 above)
Comment (lydemann) [reply libs/designsystem/src/lib/helpers/color-helper.ts:151]: I don't think we can generate an enum from the sass to ts tool.

Comment (lydemann) [reply libs/designsystem/src/lib/helpers/color-helper.ts:151]: String literal is an option

**Review by jakobe — APPROVED** [review]
giphub

**Comment by jakobe** [comment]
Closes #626

**Pull request merged**